### ISSUE TITLE
missing 0th hour when called with TimeZone in cron.v2

### ISSUE THREAD

**Issue by sairam** [robfig/cron#81]
This a bug with v2 branch

I tried to see if this is related to #55 , but its mentioned that the code is fixed. v2 diverged from master.

**Is there a plan to merge v2 with master?** 

```go
import "gopkg.in/robfig/cron.v2"
func main() {
	c := cron.New()
	c.Start()
	weekday := "0,1,2,3,4,5,6"
	hour := "0,1,2,3,4,5,6,7,8,9,10" // ,11,12,13,16,17,18,19,20,21,22,23 // hour does not work as expected
	tzName := "Asia/Kolkata"
	cron1 := fmt.Sprintf("TZ=%s 0 0 %s * * %s", tzName, hour, weekday)
	id, _ := c.AddFunc(cron1, func() { fmt.Println("Func to call") })
	entry := c.Entry(id)

	t := entry.Next
	fmt.Println(t)
	for i := 0; i < 24; i++ {
		t = entry.Schedule.Next(t)
		fmt.Println(t)
	}
}
```
Output is missing `00:00:00 from 9th Dec`.
Current Date: `Thu Dec  8 19:02:53 IST 2016`

```
2016-12-09 01:00:00 +0530 IST
2016-12-09 02:00:00 +0530 IST
2016-12-09 03:00:00 +0530 IST
2016-12-09 04:00:00 +0530 IST
2016-12-09 05:00:00 +0530 IST
2016-12-09 06:00:00 +0530 IST
2016-12-09 07:00:00 +0530 IST
2016-12-09 08:00:00 +0530 IST
2016-12-09 09:00:00 +0530 IST
2016-12-09 10:00:00 +0530 IST
2016-12-10 00:00:00 +0530 IST
2016-12-10 01:00:00 +0530 IST
2016-12-10 02:00:00 +0530 IST
2016-12-10 03:00:00 +0530 IST
2016-12-10 04:00:00 +0530 IST
2016-12-10 05:00:00 +0530 IST
2016-12-10 06:00:00 +0530 IST
2016-12-10 07:00:00 +0530 IST
2016-12-10 08:00:00 +0530 IST
2016-12-10 09:00:00 +0530 IST
2016-12-10 10:00:00 +0530 IST
2016-12-11 00:00:00 +0530 IST
2016-12-11 01:00:00 +0530 IST
2016-12-11 02:00:00 +0530 IST
2016-12-11 03:00:00 +0530 IST
```

works fine without timezone set. master does not have code for TZ . Fails when set.

```go
import 	"github.com/robfig/cron"
func main() {
	c := cron.New()
	c.Start()
	weekday := "0,1,2,3,4,5,6"
	hour := "0,1,2,3,4,5,6,7,8,9,10" // ,11,12,13,16,17,18,19,20,21,22,23 // hour does not work as expected
	// tzName := "Asia/Kolkata"
	cron1 := fmt.Sprintf("0 0 %s * * %s", hour, weekday)
	c.AddFunc(cron1, func() { fmt.Println("Func to call") })

	fmt.Println(len(c.Entries()))
	entry := c.Entries()[0]

	t := entry.Next
	fmt.Println(t)
	for i := 0; i < 24; i++ {
		t = entry.Schedule.Next(t)
		fmt.Println(t)
	}

}
```
Output:

```
2016-12-09 00:00:00 +0530 IST
2016-12-09 01:00:00 +0530 IST
2016-12-09 02:00:00 +0530 IST
2016-12-09 03:00:00 +0530 IST
2016-12-09 04:00:00 +0530 IST
2016-12-09 05:00:00 +0530 IST
2016-12-09 06:00:00 +0530 IST
2016-12-09 07:00:00 +0530 IST
2016-12-09 08:00:00 +0530 IST
2016-12-09 09:00:00 +0530 IST
2016-12-09 10:00:00 +0530 IST
2016-12-10 00:00:00 +0530 IST
2016-12-10 01:00:00 +0530 IST
2016-12-10 02:00:00 +0530 IST
2016-12-10 03:00:00 +0530 IST
2016-12-10 04:00:00 +0530 IST
2016-12-10 05:00:00 +0530 IST
2016-12-10 06:00:00 +0530 IST
2016-12-10 07:00:00 +0530 IST
2016-12-10 08:00:00 +0530 IST
2016-12-10 09:00:00 +0530 IST
2016-12-10 10:00:00 +0530 IST
2016-12-11 00:00:00 +0530 IST
2016-12-11 01:00:00 +0530 IST
2016-12-11 02:00:00 +0530 IST
```

**Comment by sairam** [comment]
if `c.Start()` is done after `AddFunc()`, it works fine.
```
	id, _ := c.AddFunc(cron1, func() { fmt.Println("Func to call") })
	entry := c.Entry(id)
	c.Start()
```
the example shows that I can run c.Start() before or after

**Comment by sairam** [comment]
Figured out the problem to
https://github.com/robfig/cron/blob/v2/spec.go#L117
It should be this way - https://github.com/robfig/cron/blob/master/spec.go#L111

instead of  `t = t.Truncate(time.Hour)`, it should have been `t = time.Date(t.Year(), t.Month(), t.Day(), t.Hour(), 0, 0, 0, t.Location())`

I think its related to the TZ being at 30 minutes offset. 

Would be great to understand when master will merge with v2 to get a v3.

**Comment by sairam** [comment]
Summary - Code got fixed via #43
